init.lua

```diff
@@ -620,17 +620,19 @@ function lsp.add_server(options)
     return false
   end
 
-  -- some lsp servers may be installed with different binary names
-  -- so if command name is a list, search for one that exists
-  if type(options.command[1]) == "table" then
-    options.command[1] = util.get_best_executable(options.command[1])
-  end
-
   -- On Windows using cmd.exe allows us to take advantage of its ability to run
   -- the correct executable, as well as running scripts.
   if PLATFORM == "Windows" and not options.windows_skip_cmd then
-    table.insert(options.command, 1, "/C")
-    table.insert(options.command, 1, "cmd.exe")
+    local escaped_commands = { }
+    if type(options.command) == "string" then
+      options.command = { options.command }
+    end
+    -- We need to escape `"` as `"""`
+    for _, v in ipairs(options.command) do
+      table.insert(escaped_commands, '"' .. string.gsub(v, '"', '"""') .. '"')
+    end
+    -- The result should be something like `cmd.exe /C ""first" "second" "third""`
+    options.command = 'cmd.exe /C "' .. table.concat(escaped_commands, " ") .. '"'
```

Comment:
Based on https://ss64.com/nt/cmd.html, you can use `/S` to simply the quoting.
`/S` simply strips the first and last quote from the string and treats that as a command,
so you need not bother yourself with yet another layer of escaping.

Otherwise the escaping looks correct to me.